streams/src/test/java/org/apache/kafka/streams/state/internals/TimestampedKeyValueStoreBuilderTest.java

```diff
@@ -0,0 +1,143 @@
+            Serdes.String(),
+            Serdes.String(),
+            new MockTime()
+        );
+    }
+
+    @Test
+    public void shouldHaveMeteredStoreAsOuterStore() {
+        final TimestampedKeyValueStore<String, String> store = builder.build();
+        assertThat(store, instanceOf(MeteredTimestampedKeyValueStore.class));
+    }
+
+    @Test
+    public void shouldHaveChangeLoggingStoreByDefault() {
+        final TimestampedKeyValueStore<String, String> store = builder.build();
+        assertThat(store, instanceOf(MeteredTimestampedKeyValueStore.class));
+        final StateStore next = ((WrappedStateStore) store).wrappedStore();
+        assertThat(next, instanceOf(ChangeLoggingTimestampedKeyValueBytesStore.class));
+    }
+
+    @Test
+    public void shouldNotHaveChangeLoggingStoreWhenDisabled() {
+        final TimestampedKeyValueStore<String, String> store = builder.withLoggingDisabled().build();
+        final StateStore next = ((WrappedStateStore) store).wrappedStore();
+        assertThat(next, CoreMatchers.equalTo(inner));
+    }
+
+    @Test
+    public void shouldHaveCachingStoreWhenEnabled() {
+        final TimestampedKeyValueStore<String, String> store = builder.withCachingEnabled().build();
+        final StateStore wrapped = ((WrappedStateStore) store).wrappedStore();
+        assertThat(store, instanceOf(MeteredTimestampedKeyValueStore.class));
+        assertThat(wrapped, instanceOf(CachingTimestampedKeyValueStore.class));
+    }
+
+    @Test
+    public void shouldHaveChangeLoggingStoreWhenLoggingEnabled() {
+        final TimestampedKeyValueStore<String, String> store = builder
+                .withLoggingEnabled(Collections.emptyMap())
+                .build();
+        final StateStore wrapped = ((WrappedStateStore) store).wrappedStore();
+        assertThat(store, instanceOf(MeteredTimestampedKeyValueStore.class));
+        assertThat(wrapped, instanceOf(ChangeLoggingTimestampedKeyValueBytesStore.class));
+        assertThat(((WrappedStateStore) wrapped).wrappedStore(), CoreMatchers.equalTo(inner));
+    }
+
+    @Test
+    public void shouldHaveCachingAndChangeLoggingWhenBothEnabled() {
+        final TimestampedKeyValueStore<String, String> store = builder
+                .withLoggingEnabled(Collections.emptyMap())
+                .withCachingEnabled()
+                .build();
+        final WrappedStateStore caching = (WrappedStateStore) ((WrappedStateStore) store).wrappedStore();
+        final WrappedStateStore changeLogging = (WrappedStateStore) caching.wrappedStore();
+        assertThat(store, instanceOf(MeteredTimestampedKeyValueStore.class));
+        assertThat(caching, instanceOf(CachingTimestampedKeyValueStore.class));
+        assertThat(changeLogging, instanceOf(ChangeLoggingTimestampedKeyValueBytesStore.class));
+        assertThat(changeLogging.wrappedStore(), CoreMatchers.equalTo(inner));
+    }
+
+    @SuppressWarnings("all")
+    @Test(expected = NullPointerException.class)
```

Comment:
nit: I thought for new tests with expected failures we used the convention of 
```java
try {
     ... some code with failure...
     fail(...)
} catch (Exception expected) {
.....
```
It's a minor point just asking and I'm with this as is.

Reply:
I think for single line tests, it's ok to use the annotation.